### DISCUSSION TITLE
How to integrate Disqus on my Statiq blog

### DISCUSSION THREAD

**Discussion by Inerska** [statiqdev/Discussions#83] [Q&A]
Hello, the question is the title, how do I integrate Disqus into my Statiq blog ? 
I've written the same code as yours (https://github.com/Inerska/inerska.github.io/blob/master/MyBlog/input/_partials/_post_prefooter.cshtml)

But I've also a JS error when I'm checking the console debugging part
![image](https://user-images.githubusercontent.com/50674490/123694031-1622cb00-d859-11eb-9447-089d7fbe0591.png)

Thanks in advance

**Comment by TheBrambleShark** [comment]
I've dug into a bit of the code for embed.js. Looks like they have updated so as to specifically expect a variable called `disqus_config` rather than the loose `disqus_` configuration variables. I also changed the value of what used to be the `disqus_url` property to point to the current document link, which is what I think you were intending in the first place. If not, you can always change it back.

Under your section, add a new div with the id `disqus_thread` (otherwise disqus will not have any place to actually build the comment thread), then replace your configuration section with this:

```html
var disqus_config = function() {
    this.page.identifier = '@Model.Destination.FileNameWithoutExtension.FullPath';
    this.page.title = '@Model.GetTitle()';
    this.page.url = '@Document.GetLink(true)';
    // this.page.category_id = '';
}
var disqus_shortname = 'agdl-dev'; // required: replace example with your forum shortname
```

It may be a good idea to put in a noscript tag at the bottom, just in case.

```html
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript" rel="nofollow">comments powered by Disqus.</a></noscript>
```

After this, it should function well. I got an error from disqus, but I suspect it's because it's running from localhost and not your domain name.

**Reply by Inerska** [reply]
I have edited my code with yours, but I still have this error message.
![image](https://user-images.githubusercontent.com/50674490/124228113-fe26a200-db0b-11eb-9db1-ffbf0b36643a.png)

https://github.com/Inerska/inerska.github.io/blob/master/MyBlog/input/_partials/_post_prefooter.cshtml

**Reply by TheBrambleShark** [reply]
> 
> 
> I have edited my code with yours, but I still have this error message.
> ![image](https://user-images.githubusercontent.com/50674490/124228113-fe26a200-db0b-11eb-9db1-ffbf0b36643a.png)
> 
> https://github.com/Inerska/inerska.github.io/blob/master/MyBlog/input/_partials/_post_prefooter.cshtml

I was getting that too. I suspect that's because your Disqus instance is expecting `agdl.dev` but instead it's being loaded on `localhost`. You may need to add that as a whitelisted site in your Disqus instance, though I'm not at all familiar with that system.

**Reply by Inerska** [reply]
> 
> 
> > I have edited my code with yours, but I still have this error message.
> > ![image](https://user-images.githubusercontent.com/50674490/124228113-fe26a200-db0b-11eb-9db1-ffbf0b36643a.png)
> > https://github.com/Inerska/inerska.github.io/blob/master/MyBlog/input/_partials/_post_prefooter.cshtml
> 
> I was getting that too. I suspect that's because your Disqus instance is expecting `agdl.dev` but instead it's being loaded on `localhost`. You may need to add that as a whitelisted site in your Disqus instance, though I'm not at all familiar with that system.

But I'm loading it in https://agdl.dev/posts/modern-cmacro-generict
At the footer you can see the Disqus error message

**Reply by daveaglick** [reply]
Thanks for jumping in here @foxtrek64 and trying to help! I guess the good news here is that the Disqus JavaScript is clearly loading (because that's where the error message is coming from). So it's some sort of client-side configuration error. Do you see any additional information in the console in devtools? I've been away with limited access this week, but can take a look early next week as an extra set of eyes to see if I notice anything amiss - it's been a while since I set up Disqus on my own blog, but as far as I know it's still working so that's another point of comparison.

**Reply by Inerska** [reply]
> 
> 
> Thanks for jumping in here @Foxtrek64 and trying to help! I guess the good news here is that the Disqus JavaScript is clearly loading (because that's where the error message is coming from). So it's some sort of client-side configuration error. Do you see any additional information in the console in devtools? I've been away with limited access this week, but can take a look early next week as an extra set of eyes to see if I notice anything amiss - it's been a while since I set up Disqus on my own blog, but as far as I know it's still working so that's another point of comparison.

![image](https://user-images.githubusercontent.com/50674490/125275434-4bc2bc00-e30f-11eb-82da-09a83465e222.png)
No any errors in the console.

![image](https://user-images.githubusercontent.com/50674490/125276963-28007580-e311-11eb-916a-b8fe50f38266.png)
And my configuration page should work

**Reply by Inerska** [reply]
Bump, sorry
